lib/shared/sdk/usbboot.js

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright 2017 resin.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * This work is heavily based on https://github.com/raspberrypi/usbboot
+ * Copyright 2016 Raspberry Pi Foundation
+ */
+
+'use strict'
+
+const _ = require('lodash')
+const Bluebird = require('bluebird')
+const usb = require('./utils/usb')
+
+/**
+ * @summary Vendor ID of "Broadcom Corporation"
+ * @type {Number}
+ * @constant
+ */
+const USB_VENDOR_ID_BROADCOM_CORPORATION = 0x0a5c
+
+/**
+ * @summary List of usbboot capable devices
+ * @type {Object[]}
+ * @constant
+ */
+const USBBOOT_CAPABLE_USB_DEVICES = [
+
+  // BCM2835
+
+  {
+    vendorID: USB_VENDOR_ID_BROADCOM_CORPORATION,
+    productID: 0x2763
+  },
+
+  // BCM2837
+
+  {
+    vendorID: USB_VENDOR_ID_BROADCOM_CORPORATION,
+    productID: 0x2764
+  }
+
+]
+
+/**
+ * @summary The number of USB endpoint interfaces in devices with a BCM2835 SoC
+ * @type {Number}
+ * @constant
+ */
+const USB_ENDPOINT_INTERFACES_SOC_BCM2835 = 1
+
+/**
+ * @summary Check if a node-usb device object is usbboot-capable
```

Comment:
All the other doc-strings just talk about 'USB devices', so perhaps `node-usb device` here should just be changed to `USB device` ?  (the mention of `node-usb` looks a bit out of place otherwise, IMHO)